skimage/transform/_geometric.py

```diff
@@ -7,6 +7,24 @@
                              _to_ndimage_mode)
 
 
+def _affine_matrix_from_vector(v):
+    """Affine matrix from linearized (d, d + 1) matrix entries."""
+    nparam = v.size
+    # solve for d in: d * (d - 1) = nparam
+    d = (1 + np.sqrt(1 + 4 * nparam)) / 2 - 1
+    dimensionality = int(d)
+    if d != dimensionality:
+        raise ValueError('Invalid number of elements for '
+                         f'linearized matrix: {nparam}')
+    part_matrix = np.reshape(
+        v, (dimensionality, dimensionality + 1)
+    )
+    matrix = np.concatenate(
+        (part_matrix, np.eye(dimensionality + 1)[-1:]), axis=0
```

Comment:
Easier to just pop the one into place?

Reply:
:shrug: The alternative code is:

```python
matrix = np.concatenate(
    (part_matrix, np.zeros(dimensionality + 1), axis=0
)
matrix[-1, -1] = 1
```

So, you get an extra line of code, for something maybe *marginally* clearer? I agree that what I'm after really is the vector commonly called e_i. But I don't think that exists in NumPy. =)

Reply:
It actually exists: what you are looking for is `np.eye(1, dimensionality + 1, dimensionality)` :wink:
```python
>>> np.eye(1, 4, 3)
array([[0., 0., 0., 1.]])
```